kodilla-good-patterns/src/main/java/com/kodilla/good/patterns/food2door/OrderProcessor.java

```diff
@@ -11,6 +15,17 @@ public OrderProcessor(InformationService informationService, OrderService orderS
     }
 
     public OrderDeliveryDto process(OrderRequest orderRequest) {
+        List<Shop> shops = orderRequest.getQuantityOfOrderedProducts().keySet().stream()
+                .map(Product::getShop)
+                .distinct()
+                .collect(Collectors.toList());
+        for (Shop shop : shops) {
+            Map<Product, Integer> shopProducts = orderRequest.quantityOfOrderedProducts.entrySet().stream()
+                    .filter(entry -> entry.getKey().getShop().equals(shop))
+                    .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+            shop.order(shopProducts);
+        }
+
         boolean isOrdered = orderService.order(orderRequest.getProvider());
         if (isOrdered) {
             informationService.inform(orderRequest.getProvider());
```

Comment:
Tutaj zgodnie z naszą rozmową - trzeba zmienić metodę `inform` tak, aby przyjmowała sensowne informacje (np. użytkownik i zamówione produkty). W tym celu zapewne musisz rozbudować klasę `OrderRequest`. Zrób to zgodnie z własną inwencją :)